tests/test_cli.py

```diff
@@ -192,48 +172,42 @@ def test_triton_profile(self, mocker, monkeypatch):
         mock_run.assert_called_once_with(["genai-perf", "-m", "add_sub"], check=True)
 
     @pytest.mark.parametrize("model", ["mock_llm"])
-    def test_triton_metrics(self, model, setup_and_teardown):
+    def test_triton_metrics(self, model):
         # Import the Model Repo
-        pid = utils.run_server(repo=MODEL_REPO)
-        setup_and_teardown.pid = pid
-        utils.wait_for_server_ready()
-
-        metrics_before = self._metrics()
+        with utils.ScopedTritonServer(repo=MODEL_REPO):
+            metrics_before = self._metrics()
 
-        # Before Inference, Verifying Inference Count == 0
-        for loaded_models in metrics_before["nv_inference_request_success"]["metrics"]:
-            if loaded_models["labels"]["model"] == model:  # If mock_llm
-                assert loaded_models["value"] == 0
+            # Before Inference, Verifying Inference Count == 0
+            for loaded_models in metrics_before["nv_inference_request_success"][
+                "metrics"
+            ]:
+                if loaded_models["labels"]["model"] == model:  # If mock_llm
+                    assert loaded_models["value"] == 0
 
-        # Model Inference
-        self._infer(model, prompt=PROMPT)
+            # Model Inference
+            self._infer(model, prompt=PROMPT)
 
-        metrics_after = self._metrics()
+            metrics_after = self._metrics()
 
-        # After Inference, Verifying Inference Count == 0
-        for loaded_models in metrics_after["nv_inference_request_success"]["metrics"]:
-            if loaded_models["labels"]["model"] == model:  # If mock_llm
-                assert loaded_models["value"] == 1
+            # After Inference, Verifying Inference Count == 0
```

Comment:
Inference count should be one after inference?

Reply:
Thanks for the Catch!